Simplified/Audiobooks/LCP/LCPAudiobooks.swift

```diff
@@ -0,0 +1,112 @@
+import NYPLAudiobookToolkit
+
+/// LCP Audiobooks helper class
+@objc class LCPAudiobooks: NSObject {
+    
+  private let audiobookUrl: URL
+  private let lcpService = LCPLibraryService()
+
+  /// LCP license object for decryption
+  private var lcpLicense: DRMLicense?
+
+  /// .lcpa archive container
+  private let container: Container
+
+  /// Distributor key - one can be found in `NYPLBook.distributor` property
+  @objc static let distributorKey = "lcp"
+  
+  /// Initialize for an LCP audiobook
+  /// - Parameter audiobookUrl: must be a file with `.lcpa` extension
+  @objc init?(for audiobookUrl: URL) {
+    self.audiobookUrl = audiobookUrl
+    do {
+      guard let (pubBox, _) = try Publication.parse(at: self.audiobookUrl) else {
+        return nil
+      }
+      self.container = pubBox.associatedContainer
+    } catch {
+      NYPLErrorLogger.logError(error, summary: "Error opening audiobook")
+      return nil
+    }
+  }
+  
+  /// Content dictionary for `AudiobookFactory`
+  @objc func contentDictionary() -> NSDictionary? {
+    let manifestPath = "manifest.json"
+    do {
+      // Relative path inside the audiobook
+      let data = try container.data(relativePath: manifestPath)
+      let publicationObject = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as? NSDictionary
+      return publicationObject
+    } catch {
+      NYPLErrorLogger.logError(error, summary: "Error reading audiobook \(manifestPath)")
+      return nil
+    }
+  }
+
+}
+
+/// DRM Decryptor for LCP audiobooks
+extension LCPAudiobooks: DRMDecryptor {
+
+  /// Decrypt protected file
+  /// - Parameters:
+  ///   - url: encrypted file URL.
+  ///   - resultUrl: URL to save decrypted file at.
+  ///   - completion: decryptor callback with optional `Error`.
+  func decrypt(url: URL, to resultUrl: URL, completion: @escaping (Error?) -> Void) {
+    loadLicense { error in
+      if let error = error {
+        completion(error)
+        return
+      }
+      guard let license = self.lcpLicense else {
+        completion(nil)
+        return
+      }
+      do {
+        let audiofileContent = try self.container.data(relativePath: url.path)
+        let data = try license.decipher(audiofileContent)
+        try data?.write(to: resultUrl, options: .atomic)
+        completion(nil)
+      } catch {
+        completion(error)
+        return
+      }
+    }
+  }
+  
+  /// Load `DRMLicense` license for audiobook once
+  /// - Parameter completion: `LCPError`, if any
+  private func loadLicense(completion: @escaping (_ error: Error?) -> Void) {
+    // If the license has already been loaded, just call back
+    if self.lcpLicense != nil {
+      completion(nil)
+      return
+    }
+    // Load publication to get license
+    let drm = DRM(brand: .lcp)
+    lcpService.loadPublication(at: audiobookUrl, drm: drm) { result in
+      switch result {
+      case .success(let updatedDrm):
+        self.lcpLicense = updatedDrm?.license
```

Comment:
there's a possible retain cycle here depending on lcpService manages the closure. Can you fix (if needed) and also make sure this is documented on the LCPLibraryService class?

Reply:
also there are some multithreading issues I believe on the `lcpLicense` property, since `loadLicense` can be invoked asynchronously by `decrypt`. Can we just pass the `lcpLicense` as a completion param instead?

Reply:
Multithreading can be an issue here... I don't think keeping caching the license worth the trouble, changed the code to read the license for each decrypt call (LCP saves the license to the book when we open it)